modules/nw-ingress-route-secret-load-external-cert.adoc

```diff
@@ -0,0 +1,76 @@
+// Module included in the following assemblies:
+//
+// * ingress/routes.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="nw-ingress-route-secret-load-external-cert_{context}"]
+= Creating route with externally managed certificate
+
+:FeatureName: Securing route with external certificates in TLS secrets
+include::snippets/technology-preview.adoc[]
+
+You can configure OpenShift routes with third-party certificate management solutions by using the `.spec.tls.externalCertificate` field of the route API. You can reference externally managed TLS certificates via secrets, eliminating the need for manual certificate management. Using the externally managed certificate reduces errors and provides faster updates. 
+
+[NOTE]
+====
+This feature applies to both edge routes and re-encrypt routes.
+====
+
+.Prerequisites
+
+* You must enable the `RouteExternalCertificate` feature gate.
+* You must have the `create` and `update` permissions on the `routes/custom-host` sub-resource.
+* You must have a secret containing a valid certificate/key pair in PEM-encoded format of type `kubernetes.io/tls`, which includes both `tls.key` and `tls.crt` keys.
+* You must place the referenced secret in the same namespace as the route you want to secure.
+
+.Procedure
+
+. Create a `role` in the same namespace as the secret to allow the router service account read access.
++
+[source,terminal]
+----
+$ oc create role secret-reader --verb=get,list,watch --resource=secrets --resource-name=<secret-name> \ <1> 
+--namespace=<current-namespace> <2>
+----
+<1> Replace <secret-name> with the actual name of your secret.
+<2> Replace <current-namespace> with the namespace where both your secret and route reside.
+
+. Create a `rolebinding` in the same namespace as the secret and bind the router service account to the newly created role.
++
+[source,terminal]
+----
+$ oc create rolebinding secret-reader-binding --role=secret-reader --serviceaccount=openshift-ingress:router --namespace=<current-namespace> <1>
+----
+<1> Replace <current-namespace> with the namespace where both your secret and route reside.
+
+. Create a `route` specifying the `.spec.tls.externalCertificate` field to reference the secret containing your certificate:
```

Comment:
I would recommend changing this step so that it is `. Create a YAML file that defines the route and specifies the secret containing your certificate using the following example`. Or something along those lines. I would have them create the YAML file because I'm guessing there might be a missing step where the file or YAML needs to be applied too. `oc apply -f <yaml_name>`.  I'm not an engineer though so I would check with devs on this.

Reply:
That sounds perfect to me. Users need to apply the yaml after defining it. :+1:

Reply:
@chiragkyal Do we need to provide any specific instruction to do the apply yaml step?

Reply:
Yeah something like

- Create a route resource by running the following command:
`oc apply -f <file.yaml>`
where `<file.yaml>` is your YAML file created above.

Feel free to change as per guidelines.